File: Study Plan/Algorithm I/Power_of_Two.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        bool flag = false;
        if(n == 0)
            return false;
        
        while(n){
            if((n & 1) == 1 and flag)
                return false;
            
            else if((n & 1) == 1)
                flag = true;
            
            n = n>> 1;
        }
        return true;
    }
};

File: Study Plan/Algorithm I/Power_of_Two.py
class Solution:
	def isPowerOfTwo(self, n: int) -> bool:
		if n <= 0:   return False
		if n == 1:   return True

		for i in range(n):
			if n % 2 == 0 and n != 2:  
				n = n / 2
			elif n == 2:  
				return True
			else:        
				return False
		return True
        